lib/salus/scanners/gosec.rb

```diff
@@ -9,7 +9,17 @@ def run
       # Shell Instructions:
       #   - -fmt=json for JSON output
       #   - gosec can scan go modules as of 2.0.0.
-      shell_return = run_shell("gosec -fmt=json #{@repository.path_to_repo}")
+      shell_return = Dir.chdir(@repository.path_to_repo) { run_shell("gosec -fmt=json ./...") }
+
+      # This produces no JSON output so must be checked before parsing stdout
+      if shell_return.stdout.blank? && shell_return.stderr.include?('No packages found')
```

Comment:
can you add a test?

Reply:
Yeah, the blank project covers this case. It's what found the issue in the first place

Reply:
one with a go project that requires a recursive scan?

Reply:
That makes sense. I have added it